### PR TITLE
easystroke: init at 0.6.0-9-gf7c1614

### PR DESCRIPTION
**Pull request by psyky**
###### Motivation for this change
easystroke: init at 0.6.0-9-gf7c1614

Easystroke is a X11 gesture recognition application.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/easystroke/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  nativeBuildInputs = [ pkgconfig gcc7 wrapGAppsHook help2man ];
+  buildInputs = [ boost dbus_glib gnome3.gtkmm xorg.libXtst xorg.inputproto 
+    xorg.xorgserver intltool libsigcxx gnome3.defaultIconTheme
+  ]; 
+
+  enableParallelBuild = true; 
+
+  patches = [ 
+    (fetchpatch {
```
Comment (andir) [inline pkgs/applications/misc/easystroke/default.nix]: Can you explain why those patches are required? Did you attempt to submit them to upstream if they are an essential part of the package?

File: pkgs/applications/misc/easystroke/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  src = fetchFromGitHub {
+    owner = "thjaeger";
+    repo = "easystroke";
+    rev = "f7c1614004e9c518bd4f6f4b3a2ddaf23911a5ef";
+    sha256 = "0map8zbnq993gchgw97blf085cbslry2sa3z4ambdcwbl0r9rd6x";
+  };
+
+  nativeBuildInputs = [ pkgconfig gcc7 wrapGAppsHook help2man ];
+  buildInputs = [ boost dbus_glib gnome3.gtkmm xorg.libXtst xorg.inputproto
```
Comment (andir) [inline pkgs/applications/misc/easystroke/default.nix]: Can you fix the indention & order here?

Sorting dependencies alphabetically and having a common indention level across all list elements is what most people seem to do. Unfortunately, there is no style guide (yet).

File: pkgs/applications/misc/easystroke/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{ stdenv, fetchFromGitHub, fetchpatch, pkgconfig, boost, dbus_glib, gnome3,
+ xorg, help2man, intltool, libsigcxx, gcc7, wrapGAppsHook}: 
+
+with stdenv.lib;
+
+let
+  version = "0.6.0-9-gf7c1614";
+in stdenv.mkDerivation {
+  name = "easystroke-${version}";
```
Comment (andir) [inline pkgs/applications/misc/easystroke/default.nix]: According to our naming guide (https://nixos.org/nixpkgs/manual/#sec-package-naming) the package name should probably be (final form after evaluation): `easystroke-unstable-2000-01-31` (with correct date)

File: pkgs/applications/misc/easystroke/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{ stdenv, fetchFromGitHub, fetchpatch, pkgconfig, boost, dbus_glib, gnome3,
```
Comment (andir) [inline pkgs/applications/misc/easystroke/default.nix]: Why did you specify `gcc7` there (& within nativeBuildInputs)? Usually sticking with just using whatever `stdenv` provides is what you want.

**Review by andir — CHANGES_REQUESTED** [review]
Thanks :+1:

I left some comments in the review.

**Comment by andir** [comment]
@GrahamcOfBorg build easystroke

**Comment by timokau** [comment]
Replaced by https://github.com/NixOS/nixpkgs/pull/52424.

**Pull request closed without merging**